solana/programs/example-native-token-transfers/src/instructions/admin.rs

```diff
@@ -154,6 +156,167 @@ pub fn claim_ownership(ctx: Context<ClaimOwnership>) -> Result<()> {
     )
 }
 
+// * Set token authority
+
+#[derive(Accounts)]
+pub struct SetTokenAuthority<'info> {
+    #[account(
+        has_one = owner,
+        constraint = config.paused @ NTTError::NotPaused,
+    )]
+    pub config: Account<'info, Config>,
+
+    pub owner: Signer<'info>,
+
+    #[account(
+        mut,
+        address = config.mint,
```

Comment:
should also be checked via a `has_one` constrained of the `config` account for consistency.

Reply:
Would this not be redundant checks as we are checking the equality both ways?

I think the repo has cases of `address = config.mint` without `has_one` in [broadcast_id](https://github.com/wormhole-foundation/native-token-transfers/blob/main/solana/programs/example-native-token-transfers/src/transceivers/wormhole/instructions/broadcast_id.rs#L15), [release_inbound](https://github.com/wormhole-foundation/native-token-transfers/blob/main/solana/programs/example-native-token-transfers/src/instructions/release_inbound.rs#L39), etc. so I went with this check here. If there's a constraint preference in using `address` over `has_one` or vice versa, we can stick to that.

Reply:
Sorry, my comment was imprecise. My "also" was not meant as "in addition to" but as in "also via `has_one` like the `owner`". Right now checking the `owner` one way and the `mint` another seems strange. I think that the `address` check has the advantage that it actually is located with the account that is being checked, while the `has_one` is more terse. So either both should use `has_one` or neither, I'd argue.

Reply:
That makes sense. For now, I'll arbitrarily choose the `has_one` constraint over the `address` and keep that the case for this file at least.